lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -137,15 +137,31 @@ def store_attributes
       end
 
       def descendants(filters = {})
-        Assignment.descendants_of(self).where(filters)
+        if filters.empty?
+          Assignment.descendants_of(self)
+        else
+          valid_filters = PolicyElement.columns_hash.map { |column| column.first.to_sym }
+          filters.each_key do |key|
+            raise(ArgumentError, "#{key} is not a valid filter") unless valid_filters.include?(key)
+          end
+          Assignment.descendants_of(self).where(filters)
+        end
       end
 
       def link_descendants
         LogicalLink.descendants_of(self)
       end
 
       def ancestors(filters = {})
-        Assignment.ancestors_of(self).where(filters)
+        if filters.empty?
```

Comment:
I thought empty params were because Ruby doesn't support method overloading.

In either case, I checked the `where` implementation and it just does the same check on its end (for empty args), and returns self. I'm fine removing the check here and delegating